### PR TITLE
lantiq: remove bogus PCI ID + NVMEM

### PR DESCRIPTION
**Pull request by neheb**
ping @hauke @robimarko @lmadam20 @xdarklight

### CODE REVIEW

**Comment by neheb** [comment]
Another example of this is: https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/qca9563_netgear_wndr.dtsi#L274

I guess the compatible string really does override.

File: package/kernel/mac80211/patches/ath9k/513-ath9k_add_pci_ids.patch [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-@@ -662,6 +662,7 @@ int ath9k_hw_init(struct ath_hw *ah)
- 
- 	/* These are all the AR5008/AR9001/AR9002/AR9003 hardware family of chipsets */
- 	switch (ah->hw_version.devid) {
-+	case AR9300_DEVID_INVALID:
- 	case AR5416_DEVID_PCI:
- 	case AR5416_DEVID_PCIE:
- 	case AR5416_AR9100_DEVID:
---- a/drivers/net/wireless/ath/ath9k/hw.h
-+++ b/drivers/net/wireless/ath/ath9k/hw.h
```
Comment (hauke) [inline package/kernel/mac80211/patches/ath9k/513-ath9k_add_pci_ids.patch:12]: There was a discussion about this PCI ID in this patch already:
https://patchwork.ozlabs.org/project/openwrt/patch/20220116213025.1379508-1-hauke@hauke-m.de/

Comment (neheb) [reply package/kernel/mac80211/patches/ath9k/513-ath9k_add_pci_ids.patch:12]: That link is to a TD-W8970, which has

compatible = "pci168c,002e";

already.

Although the node is wrong: pci168c,002e  # AR9287 according to Documentation/devicetree/bindings/net/wireless/qca,ath9k.yaml

pci168c,0030 is the correct one.

Comment (neheb) [reply package/kernel/mac80211/patches/ath9k/513-ath9k_add_pci_ids.patch:12]: Added commit to fix this.

**Comment by neheb** [comment]
Rebased.

Sort of depends on https://github.com/openwrt/openwrt/pull/9775

**Comment by neheb** [comment]
@hauke removed the mac80211 patch. Looks like it needs to remain. Now this is only an NVMEM conversion and compatible line fix.

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**